src/methods/call.js

```diff
@@ -613,3 +674,55 @@ export function Call(realm: Realm, F: Value, V: Value, argsList?: Array<Value>):
   invariant(F.$Call, "no call method on this value");
   return F.$Call(V, argsList);
 }
+
+function isValueAnUnknownAbstractValue(val: Value): boolean {
```

Comment:
I don't understand why this function is a a good proxy for why we should inline effects. For example, why is a primitive value like "true" not getting inlined?

Reply:
BooleanValue is an instance of a ConcreteValue, so it would be inlined unless part of a conditional with an abstract value.

Reply:
Right. But your logic would still stay to not inline a conditional abstract value whose leaf values are concrete values, but you would inline an conditional abstract value where the leaf values are some other (obscure?) abstract values.

Reply:
It will inline if a conditionals leaf values are all concrete, if a single leaf is abstract it won’t though. I tried it in the reverse, but we inline and create quite a bit more bloat. Maybe I’ll refine it more.